velox/exec/tests/utils/QueryAssertions.h

```diff
@@ -98,14 +100,84 @@ class TestScopedAbortInjection {
   ~TestScopedAbortInjection();
 };
 
+class OOMInjector {
+ public:
+  OOMInjector(
+      const std::function<bool()>& oomCondition,
+      uint64_t iterationWaitMs)
+      : oomCondition_(oomCondition), iterationWaitMs_(iterationWaitMs) {}
+
+  OOMInjector(const OOMInjector& other)
+      : OOMInjector(other.oomCondition_, other.iterationWaitMs_) {}
+
+  auto& operator=(const OOMInjector& other) {
+    oomCondition_ = other.oomCondition_;
+    iterationWaitMs_ = other.iterationWaitMs_;
+    return *this;
+  }
+
+  ~OOMInjector() {
+    if (oomThread_) {
+      stop_.store(true);
+      waitForStop_.notifyAll();
+      oomThread_->join();
+    }
+  }
+
+  void start() {
+    // Since this relies on TestValue to trigger the OOMs, it only supports
+    // debug builds.
+#ifdef NDEBUG
```

Comment:
Why not put this class implementation under #ifdef NDEBUG and the production code does nothing?

Reply:
I wanted to make it clear that if you enable it in a production run it's not doing anything, otherwise it just silently looks like it's working.  Given that it's only used in testing we don't have to hide it in production runs.